### PR TITLE
Allows for some jobs & drones to use vending machines without a registered account

### PR DESCRIPTION
**Pull request by Altoids1**
### God damned Capitalism fuckin' up my wonderful, glorious Fully Automatic Luxury Gay Space Communism

image

Right now Centcomm staff and ``simple_mob``s (like drones) lack the ability to use vending machines. This is stupid, so I'm giving them authorization to just buy whatever from them for free.

#### Changelog
:cl:  Altoids
bugfix: Central Command staff, as well as simple_mobs, can use vending machines again!
/:cl:

### CODE REVIEW

**Comment by Altoids1** [comment]
I'm thinking I might make it so that ERT just have an ID card that's marked as ignoring Capitalism, instead of making them ignore Capitalism themselves, inherently. Will work on it.

**Comment by ThatLing** [comment]
Not all simple_animals should have access, that's stupid. If you want drones to have access, add them to the ai check. (do borgs even have access?)

**Comment by Moja10** [comment]
AFAIK borgs can't use vendors that cost money, they show up as inventoryless. For things that don't cost money to vend, they can use them

**Comment by MegaEmpirical** [comment]
I'd love to see this fixed at the least: https://github.com/tgstation/tgstation/issues/41637

Nich broke it and didn't care so I never made an issue for it on yogs.

**Comment by ThatLing** [comment]
@nichlas0010 lol

File: yogstation/code/modules/mob/living/silicon/silicon.dm [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /mob/living/silicon
 	var/list/law_history = list()
+	ignores_capitalism = TRUE // Yogs -- Let the robits buy me a damned drink for Christ's sake
```
Comment (Amelia0010) [inline yogstation/code/modules/mob/living/silicon/silicon.dm]: watch how one man disabled economy entirely with just one line

**Review by Amelia0010 — DISMISSED** [review]
can we not make the people who are usually forced by the rules to obey peoples' orders, bypass capitalism?

File: yogstation/code/modules/mob/living/silicon/silicon.dm (hunk 1 above)
Comment (Altoids1) [reply yogstation/code/modules/mob/living/silicon/silicon.dm]: :)

**Review by oklews — COMMENTED** [review]
Looks alright, good Idea

File: yogstation/code/modules/mob/living/silicon/silicon.dm (hunk 1 above)
Comment (Amelia0010) [inline yogstation/code/modules/mob/living/silicon/silicon.dm]: Can we not completely nullify the economy while pretending we're not?

**Review by Amelia0010 — DISMISSED** [review]

**Pull request merged**